doc/geany.txt

```diff
@@ -3357,6 +3357,10 @@ Delete to line start            Ctrl-Shift-BackSpace      Deletes from the begin
                                                           current caret position.
 
 Duplicate line or selection     Ctrl-D                    Duplicates the current line or selection.
+							  If the selection spans more than one line,
+							  this duplicates the lines containing selections.
+							  If the selection spans only a single line, then only
+							  the selection is duplicated.
 
```

Comment:
@elextr nice try, but not quite that :)  If  a portion of a single line is selected, it only duplicates that portion (e.g. if you select `second` in `select the second word` and duplicate, it gives you `select the secondsecond word`).

Basically this tries to "do the right thing":
* if you have nothing selected, it duplicates the current line
* if the selection spans a single line, it duplicates that portion
* if the selection spans more than one line, it duplicates the whole block of lines on which the selection spans (e.g. extends the selection to span whole lines, and duplicates that portion)